vacasaconnect.py

```diff
@@ -0,0 +1,243 @@
+        if headers is None:
+            headers = {}
+
+        if params is None:
+            params = {}
+
+        r = requests.get(url, headers=headers, params=params)
+        r.raise_for_status()
+
+        return r
+
+    @staticmethod
+    def _post(url, payload: dict, headers: dict = None):
+        """HTTP POST request helper"""
+        if not headers:
+            headers = {}
+
+        r = requests.post(url, data=payload, headers=headers)
+        r.raise_for_status()
+
+        return r
+
+    def get_units(self, params: dict = None, include_terminated: bool = False):
+        """Retrieve multiple units.
+
+        Args:
+            params: A dict containing a key for each query string parameter
+                with a corresponding value. See https://connect.vacasait.com/
+                for more detail.
+            include_terminated: Whether or not to include units that are
+                currently terminated or pending termination.
+
+        Yields:
+            An iterator of units. Each unit is a dict.
+        """
+        if params is None:
+            params = {}
+
+        if not include_terminated:
+            params['filter[terminated]'] = 0
+
+        url = f"{self.endpoint}/v1/units"
+        headers = self._headers()
+        more_pages = True
+
+        while more_pages:
+            r = self._get(url, headers=headers, params=params)
+            yield from r.json().get('data')
+
+            if r.json().get('links').get('next'):
+                more_pages = True
+                url = r.json().get('links').get('next')
+            else:
+                more_pages = False
+
+    def get_unit_by_id(self, unit_id: str, params: dict = None) -> dict:
+        """Retrieve a single unit by its primary identifier.
+
+        Args:
+            unit_id: The unique identifier for the individual unit.
+            params: A dict containing a key for each query string parameter
+                with a corresponding value. See https://connect.vacasait.com/
+                for more detail.
+
+        Returns:
+            A dict containing attributes about the individual unit.
+        """
+        if params is None:
+            params = {}
+
+        url = f"{self.endpoint}/v1/units/{unit_id}"
+        r = self._get(url, headers=self._headers(), params=params)
+
+        return r.json().get('data')
+
+    def get_availability(self, params: dict = None):
+        """Retrieve availabilities.
+
+        Args:
+            params: A dict containing a key for each query string parameter
+                with a corresponding value. See https://connect.vacasait.com/
+                for more detail.
+
+        Yields:
+            An iterator of availabilities. Each availability is a dict.
+        """
+        if params is None:
+            params = {}
+
+        url = f"{self.endpoint}/v1/availability"
+        headers = self._headers()
+        more_pages = True
+
+        while more_pages:
```

Comment:
If we end up with more than a couple paginated API endpoints, this might be worth DRYing up

Reply:
Agreed.